python/src/helpers/find_cell_ids_in_same_hierarchy.py

```diff
@@ -0,0 +1,37 @@
+def get_all_cell_ids_in(cell_set):
+    cell_ids = cell_set["cellIds"]
+
+    children = cell_set.get("children", None)
+    if children:
+        for child in children:
+            cell_ids += get_all_cell_ids_in(child)
+
+    return cell_ids
+
+
+def get_all_cell_ids_on_current_level(cell_sets, ignoring_set_id=None):
+    cell_sets_to_return = []
+
+    for cell_set in cell_sets:
+        if ignoring_set_id and cell_set["key"] == ignoring_set_id:
+            continue
+
+        # cell_sets_to_return.append(cell_set["key"])
+        cell_sets_to_return += get_all_cell_ids_in(cell_set)
+
+    return cell_sets_to_return
+
+
+# returns a list with
+def find_cell_ids_in_same_hierarchy(key, cell_sets):
+    for cell_set in cell_sets:
+        if cell_set["key"] == key:
+            return get_all_cell_ids_on_current_level(cell_sets, ignoring_set_id=key)
+
+        if cell_set.get("children", None):
+            result = find_cell_ids_in_same_hierarchy(key, cell_set["children"])
+
+            if result:
+                return result
```

Comment:
I think you could just `return find_cell_ids_in_same_hierarchy(key, cell_set["children"])` in here?

Reply:
Actually this is not just a recursive call, it is a recursive call on the children cell set, so one level down.

In the case it is not found in the children cell set I need the `for` to iterate over to the next cell set in the same level.
I was trying to follow the logic of `find_cells_by_set_id` that searches in a similar way:
`
        if cell_set.get("children", None):
            result = find_cells_by_set_id(needle, cell_set["children"])

            if result:
                return result
`